cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/internal/mazha/CouchClient.java

```diff
@@ -480,6 +483,15 @@ public ChangesResult changes(final Map<String, Object> options) {
         return executeToJsonObjectWithRetry(connection, ChangesResult.class);
     }
 
+    private ChangesResult changesRequestWithPost(String filter, String body, Object since, Integer limit) {
```

Comment:
For consistency with `changes(final Map<String, Object> options)`, this should have the signature `changesRequestWithPost(String filter, String body, final Map<String, Object> options` and we should hoist the call to `getParametrizedChangeFeedOptions` into the caller.

Reply:
@jjrodrig actually i just realised it can be `changesRequestWithPost(String body, final Map<String, Object> options)` and the caller can put `filter` into the `options`